amazon-s3-backup/run.sh

```diff
@@ -14,13 +19,29 @@ local_backups_to_keep="$(bashio::config 'local_backups_to_keep' '3')"
 monitor_path="/backup"
 jq_filter=".backups|=sort_by(.date)|.backups|reverse|.[$local_backups_to_keep:]|.[].slug"
 
+
 export AWS_ACCESS_KEY_ID="$(bashio::config 'aws_access_key')"
 export AWS_SECRET_ACCESS_KEY="$(bashio::config 'aws_secret_access_key')"
 export AWS_REGION="$bucket_region"
 
 bashio::log.debug "Using AWS CLI version: '$(aws --version)'"
-bashio::log.debug "Command: 'aws s3 sync $monitor_path s3://$bucket_name/ --no-progress --region $bucket_region --storage-class $storage_class'"
-aws s3 sync $monitor_path s3://"$bucket_name"/ --no-progress --region "$bucket_region" --storage-class "$storage_class"
+
+readarray -t backups < <(bashio::api.supervisor GET /backups/info false .backups | jq -c '.[]')
+for backup in "${backups[@]}"; do
+  bashio::log.debug $(echo "${backup}" | jq -C .)
+
+  slug=$(echo "${backup}" | jq -r .slug)
+  name=$(echo "${backup}" | jq -r .name)
```

Comment:
I think we should also handle special characters. See limitations for object keys:  https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html#:~:text=Characters%20that%20might%20require%20special%20handling

I suggest to use `$name` as is in the metadata and create a separate variable (for instance `$object_key`) for the object key with replaced/removed characters which need special handling and should be avoided, mentioned in the above link.
Logic could be something like:
1. Replace spaces/multiple whitespace characters with a single `_`
2. Replace all characters not mentioned [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html#:~:text=and%20other%20APIs.-,Safe%20characters,-The%20following%20character) with `-`, and with a single `-` if there are multiple consecutive unsafe characters.

What do you think?